exporter/awss3exporter/factory.go

```diff
@@ -53,13 +75,26 @@ func createLogsExporter(ctx context.Context,
 
 	s3Exporter := newS3Exporter(cfg, "logs", params)
 
-	return exporterhelper.NewLogs(ctx, params,
+	logsExporter, err := exporterhelper.NewLogs(ctx, params,
 		config,
 		s3Exporter.ConsumeLogs,
 		exporterhelper.WithStart(s3Exporter.start),
 		exporterhelper.WithQueue(cfg.QueueSettings),
 		exporterhelper.WithTimeout(cfg.TimeoutSettings),
 	)
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.OTelAttrsToS3.S3Prefix == "" {
+		return logsExporter, err
+	}
+
+	wrapped := &baseLogsExporter{
+		Component: logsExporter,
+		Logs:      batchperresourceattr.NewBatchPerResourceLogs(cfg.OTelAttrsToS3.S3Prefix, logsExporter),
```

Comment:
I assume the idea here is to optimize writes to S3 by having each of these batches based on the prefix they're aligned to? Like it.